code/modules/urist/modules/shipbattles/combatcomputer.dm

```diff
@@ -93,32 +93,69 @@
 			//maybe change passshield data to ammo?
 			data["existing_weapons"] = weapons
 
-	if(target) //come back to this when making pvp
+	if(target && istype(target, /mob/living/simple_animal/hostile/overmapship)) //We need a different UI for player ships
 		var/mob/living/simple_animal/hostile/overmapship/OM = target
-		var/integrity = (OM.health / OM.maxHealth) * 100
-		data["target"] = 1
-		data["targetname"] = OM.name
-		data["targethealth"] = integrity
-		data["targetshield"] = OM.shields
+		var/integrity = round(max((OM.health / OM.maxHealth) * 100, 0), 0.01)
+		var/maxshields = 0
 
 		for(var/datum/shipcomponents/C in OM.components)
 			var/status
 			if(C.broken)
 				status = "Broken"
-			else if(!C.broken)
+			else if(istype(C, /datum/shipcomponents/shield))
+				maxshields = C:strength
```

Comment:
Happy to change it, but I'm a little confused if it's necessary as we're already explicitly checking the type in the line above, so it would be impossible to call C:strength unless it is of type /datum/shipcomponents/shield regardless?

Reply:
It's not strictly necessary to make it work per se, but it's far better for maintaining a long-term project such as this one. Using the : operator, if the strength or overcharged vars ever changed or were removed, this function would error out, and it wouldn't be caught by the compiler as it would be using the . operator. Because SS13 codebases are these big long-term projects with multiple contributors, it's best not to do something that can potentially introduce hard to track errors down the line when there's the . operator instead (even the BYOND docs recommend against using the : operator for the same reason)

Reply:
Ahh that makes sense! Thanks for the explanation! Switched to proper type in all files in 9b46ce7fe1b09c41eea235645a4927e7ecc690c9